cmd/ocm-backplane/console/console.go

```diff
@@ -644,6 +648,35 @@ func (o *consoleOptions) printURL() error {
 	return nil
 }
 
+func (o *consoleOptions) beforeStartCleanUp(ce containerEngineInterface) error {
+	clusterID, err := getClusterID()
+	if err != nil {
+		return fmt.Errorf("error getting cluster ID: %v", err)
+	}
+	containersToCleanUp := []string{
+		fmt.Sprintf("monitoring-plugin-%s", clusterID),
+		fmt.Sprintf("console-%s", clusterID),
+	}
+
+	logger.Infoln("Starting initial cleanup of containers")
+
+	for _, c := range containersToCleanUp {
+		exist, err := ce.containerIsExist(c)
+		if err != nil {
+			return fmt.Errorf("failed to check if container %s exists: %v", c, err)
+		}
+		if exist {
+			err := ce.stopContainer(c)
+			if err != nil {
+				return fmt.Errorf("failed to stop container %s during the cleanup process: %v", c, err)
+			}
```

Comment:
I think the part is un-tested, see if you can improve the test to cover this part?

https://app.codecov.io/gh/openshift/backplane-cli/pull/584?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=openshift

Reply:
I've added a mock for the ContainerEngineInterface to be able to test a second part of the code. 
I tried to do `mockgen` to generate an auto mock but the original has internal functions only and can't be used outside the package. In order to fix it we would need to refactor entire package and make sure it won't impact something else. 

Hence from here I can see two ways to get forward:
1. Revert back to 50% coverage and merge
2. Keep as it is now with manually added mock

`
containerEngineInterface interface {
	pullImage(imageName string) error
	putFileToMount(filename string, content []byte) error
	stopContainer(containerName string) error
	// some container engines have special handlings for different types of containers
	runConsoleContainer(containerName string, port string, consoleArgs []string, envVars []envVar) error
	runMonitorPlugin(containerName string, consoleContainerName string, nginxConf string, pluginArgs []string) error
	containerIsExist(containerName string) (bool, error)

`

Reply:
Thank you for exploring this!

Let's go with "Revert back to 50% coverage and merge".  

The logic to throw an error in the code is straightforward, I think we are ok to not cover it, as the manual mocking could become another maintenance burden. The manual mocking also involved another 3rd-party package, it can be another thing to maintain.

Reply:
We can refactor the container engine interface in the future.  https://issues.redhat.com/browse/OSD-24949